src/components/screens/main-screen.tsx

```diff
@@ -231,11 +323,46 @@ const MainScreen: React.FC = () => {
   /**
    * Renders courses
    */
-  const renderCourses = () => (
-    <Stack spacing={3}>
-      {courses.map(course => <CourseCard course={course}/>)}
-    </Stack>
-  );
+  const renderCourses = () => {
+    const startIndex = (page - 1) * coursesPerPage;
+    const endIndex = startIndex + coursesPerPage;
+    const displayedCourses = courses.slice(startIndex, endIndex);
+  
+    /**
+     * Handles page change
+     */
+    const handlePageChange = (_: React.ChangeEvent<unknown>, newPage: number) => {
+      setPage(newPage);
+      // Wait for a short delay before scrolling to the first course card on the new page
+      setTimeout(() => {
+        if (courseListRef.current !== null) {
+          courseListRef.current.scrollIntoView({ behavior: "smooth" });
+        }
+      }, 50);
+    };
+  
+    return (
+      <Stack spacing={3}>
+        <div ref={courseListRef}/>
+        {displayedCourses.map(course => (
+          <CourseCard key={course.code} course={course}/>
+        ))}
+        <Stack
+          direction="row"
+          sx={{
+            display: "flex",
+            justifyContent: "center"
+          }}
+        >
+          <Pagination
+            count={Math.ceil(courses.length / coursesPerPage)}
+            page={page}
+            onChange={handlePageChange}
+          />
```

Comment:
```suggestion
          { courses.length > coursesPerPage &&
            <Pagination
              count={Math.ceil(courses.length / coursesPerPage)}
              page={page}
              onChange={handlePageChange}
            />
          }
```

Show the pagination if the course count is more than one page (Check the lint, I am not 100% sure my code style complies with the linter)